app/src/main/java/com/jerboa/ui/components/remove/comment/CommentRemoveActivity.kt

```diff
@@ -0,0 +1,95 @@
+package com.jerboa.ui.components.remove.comment
+
+import android.util.Log
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Send
+import androidx.compose.material3.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.jerboa.JerboaAppState
+import com.jerboa.R
+import com.jerboa.api.ApiState
+import com.jerboa.db.entity.isAnon
+import com.jerboa.model.AccountViewModel
+import com.jerboa.model.RemoveViewModel
+import com.jerboa.ui.components.common.ActionTopBar
+import com.jerboa.ui.components.common.getCurrentAccount
+import com.jerboa.ui.components.remove.RemoveItemBody
+import it.vercruysse.lemmyapi.v0x19.datatypes.Comment
+
+object CommentRemoveReturn {
+    const val COMMENT_VIEW = "comment-remove::return(comment-view)"
+    const val COMMENT_SEND = "comment-remove::send(comment-view)"
+}
+
+@Composable
+fun CommentRemoveActivity(
+    appState: JerboaAppState,
+    accountViewModel: AccountViewModel,
+) {
+    Log.d("jerboa", "got to create comment remove activity")
+
+    val ctx = LocalContext.current
+    val account = getCurrentAccount(accountViewModel = accountViewModel)
+
+    val removeViewModel: RemoveViewModel = viewModel()
+    val comment = appState.getPrevReturn<Comment>(key = CommentRemoveReturn.COMMENT_SEND)
```

Comment:
These appState returns... I wish there was a better way to do them, because the types seem forced. Its basically a duck-typed key-value store.

Reply:
Yeah it aint the best, rather had compile time safety but its the best we can do rn.